nshlib/nsh_alias.c

```diff
@@ -0,0 +1,441 @@
+ *
+ * Input Parameters:
+ *   vtbl - The NSH console.
+ *   argc - Amount of argument strings in command.
+ *   argv - The argument strings.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int cmd_alias(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
+{
+  FAR struct nsh_alias_s *alias;
+  FAR char **arg;
+  FAR char *value;
+  int ret;
+
+  /* Init, if necessary */
+
+  alias_init(vtbl);
+
+  if (argc < 2)
+    {
+      /* Print the alias list */
+
+      alias_printall(vtbl);
+      return OK;
+    }
+
+  /* Traverse through the argument vector */
+
+  ret = OK;
+  for (arg = argv + 1; *arg; arg++)
+    {
+      /* Look for name=value */
+
+      if ((value = strchr(*arg, '=')) != NULL)
+        {
+          /* Save new / modify existing alias */
+
+          *value++ = '\0';
+          if (alias_save(vtbl, *arg, value) < 0)
+            {
+              nsh_error(vtbl, g_savefailfmt, *arg, value);
+              ret = ERROR;
+            }
+        }
+      else if ((alias = alias_find(vtbl, *arg)) != NULL)
+        {
+          /* Found it */
+
+          nsh_output(vtbl, g_aliasfmt, alias->name, alias->value);
+        }
+      else
+        {
+          /* Nothing found */
+
+          nsh_error(vtbl, g_noaliasfmt, "alias", *arg);
+          ret = ERROR;
+        }
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: cmd_unalias
+ *
+ * Description:
+ *   Handle 'cmd_unalias' command from terminal.
+ *
+ *   Command does one of two things:
+ *   1) If the '-a' argument is given, all aliases are destroyed.
+ *   2) If a known alias name is given, the name=value pair is destroyed.
+ *
+ * Input Parameters:
+ *   vtbl - The NSH console.
+ *   argc - Amount of argument strings in command.
+ *   argv - The argument strings.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int cmd_unalias(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
+{
+  FAR struct nsh_alias_s *alias;
+  FAR char **arg;
+  int option;
+  int ret;
+
+  /* Init, if necessary */
+
+  alias_init(vtbl);
+
+  if (argc < 2)
+    {
+      /* No arguments given, this is an error */
+
+      return ERROR;
+    }
+
+  /* If '-a' is provided, then just wipe them all */
+
+  if ((option = getopt(argc, argv, "a")) != ERROR)
+    {
+      alias_removeall(vtbl);
+      return OK;
+    }
+
+  /* Traverse through the argument vector */
+
+  ret = OK;
+  for (arg = argv + 1; *arg; arg++)
+    {
+      if ((alias = alias_find(vtbl, *arg)) != NULL)
+        {
+          /* Found it */
+
+          alias_delete(vtbl, alias);
+        }
+      else
+        {
+          /* Nothing found */
+
+          nsh_error(vtbl, g_noaliasfmt, "unalias", *arg);
+          ret = ERROR;
```

Comment:
ret = -ENOENT;

Reply:
Nice catch, I'll fix this and all the rest of the return values.